adam/learner/integrated_learner.py

```diff
@@ -620,6 +636,47 @@ def update_concept_semantics(
                     self.semantics_graph.add_edge(
                         obj_con, other_con, slot=slot, weight=1.0
                     )
+                    # if the object is a wug - a new object heard through generics
+                    if not any(
+                        [
+                            obj_con.debug_string == c.debug_string
```

Comment:
Logic shouldn't be based off of the debug strings -- Why are they being referenced here?